src/sema/expression/literals.rs

```diff
@@ -799,6 +805,18 @@ pub(super) fn array_literal(
     }
 }
 
+/// When converting an array literal to slice of slices, what is the slice element of
+/// the array? (or array of array, or array of array of array etc).
```

Comment:
This is not clear to me. Can you give an example?

Reply:
You're right, this is totally not clear. I've re-written this and it's not necessary.